File: Python/imageprocessing.py
def main():
  H,W,N,M = map(int,input().split())
  before = [list(map(int,input().split())) for _ in range(H)]
  ker = [list(map(int,input().split()))[::-1] for _ in range(N)][::-1]
  image = [[0]*(W-M+1) for _ in range(H-N+1)]
  for i in range(len(image)):
    for j in range(len(image[i])):
      for a in range(len(ker)):
        for b in range(len(ker[a])):
          image[i][j] += before[i+a][j+b] * ker[a][b]

  print('\n'.join([' '.join(list(map(str, arr))) for arr in image]))







if __name__ == '__main__':
  main()

File: Java/imageprocessing.java
import java.util.*;
public class imageprocessing {
   public static void main(String[] args) {

     Scanner input = new Scanner(System.in);
     int H = input.nextInt();
     int W = input.nextInt();
     int N = input.nextInt();
     int M = input.nextInt();

     int[][] before = new int[H][W];
     for (int i = 0; i < H; i++) { 
       for (int j = 0; j < W; j++) {
         before[i][j] = input.nextInt();
       }
     }
     int[][] ker = new int[N][M];
     for (int i = N-1; i >= 0; i--) { 
       for (int j = M-1; j >= 0; j--) {
         ker[i][j] = input.nextInt();
       }
     }
     int[][] after = new int[H-N+1][W-M+1];
     for (int j = 0; j < after.length; j++) { 
       for (int i = 0; i < after[0].length; i++) {
         for (int k = 0; k < N; k++) {
           for (int l = 0; l < M; l++) {
             after[i][j] += ker[k][l] * before[i+k][j+l];
           }
         }
       }
     }

     for (int[] arr: after) {
        for (int i = 0; i < arr.length; i++) {
          System.out.print(arr[i]);
          System.out.print(" ");
        }
        System.out.println();
     }
     

    




    }


}
